File: Programmers_Algorithm/dfs&bfs/네트워크.cpp
#include <string>
#include <vector>

using namespace std;
bool visited[201];
bool dfs(int i, vector<vector<int>> computers)
{
    if(visited[i])
    {
        return false;
    }
    visited[i]=true;
    
    for(int j=0;j<computers[i].size();j++)
    {
        if(computers[i][j]==1)
        {
            dfs(j,computers);
        }
    }
    return true;
    
}
int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    for(int i=0;i<n;i++)
    {
        answer = answer + dfs(i, computers);
    }
    return answer;
}

File: Programmers_Algorithm/dfs&bfs/네트워크.py
check = [0] * 201

def dfs(i, computers):
    if check[i] == 1:
        return 0
    print(len(computers[i]))
    check[i] = 1
    for j in range(len(computers[i])):
        if computers[i][j] == 1:
            dfs(j, computers)

    return 1
    
def solution(n, computers):
    answer = 0
    
    print(computers)
    
    for i in range(len(computers[0])):
        
        answer = answer + dfs(i, computers)
    return answer